addons/portal/controllers/message_reaction.py

```diff
@@ -0,0 +1,13 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.mail.controllers import message_reaction
+from odoo.addons.portal.utils import get_portal_partner
+
+
+class MessageReactionController(message_reaction.MessageReactionController):
+
+    def _compute_reaction_author(self, thread, **kwargs):
+        partner, guest = super()._compute_reaction_author(thread, **kwargs)
+        if not partner and not guest:
+            partner = get_portal_partner(thread, kwargs.get("hash"), kwargs.get("pid"), kwargs.get("token"))
```

Comment:
Don't we want portal partner to take priority over guest?
Isn't it how we do in message post?